### PR TITLE
Use IDF's ADC Driver and Add analogReadMilliVolts

### PR DESCRIPTION
**Pull request by me-no-dev**

### CODE REVIEW

File: cores/esp32/esp32-hal-adc.c [hunk 1]
```diff
@@ -151,86 +102,106 @@ bool IRAM_ATTR __adcAttachPin(uint8_t pin){
...
-        SET_PERI_REG_MASK(SENS_SAR_MEAS_START2_REG, SENS_MEAS2_START_SAR_M);
+        r = adc2_get_raw( channel, __analogWidth, &value);
+        if ( r == ESP_OK ) {
+            return value;
+        } else if ( r == ESP_ERR_INVALID_STATE ) {
+            log_e("%s: ADC2 not initialized yet.", esp_err_to_name(r));
+        } else if ( r == ESP_ERR_TIMEOUT ) {
+            log_e("%s: ADC2 is in use by Wi-Fi.", esp_err_to_name(r));
+        } else {
+            log_e("%s", esp_err_to_name(r));
```
Comment (stickbreaker) [inline cores/esp32/esp32-hal-adc.c]: Might you want to include `pin` in the errors messages?

Comment (me-no-dev) [reply cores/esp32/esp32-hal-adc.c]: happy now? :D

**Comment by Tomat7** [comment]
Hello!

Why adcStart()/adcBusy()/adcEnd() disappeared?
What to use instead of these functions for non-blocking continues reading ADC?

Thanks.
ILYA.

**Comment by HoverClub** [comment]
I'll second the above!  Hopefully there are good reasons for removing this functionality.

**Pull request merged**